src/BreakpointViewer.cpp

```diff
@@ -0,0 +1,595 @@
+
+			switch (bp.type) {
+			case Breakpoint::BREAKPOINT: {
+				bpRow = createTableRow(BREAKPOINT);
+				setBreakpointStatus(BREAKPOINT, bpRow, true);
+
+				bool currentSlot = breakpoints->isBreakpoint(breakpoints->getAddress(index));
+				QBrush color = palette().brush(runState | currentSlot ?
+					QPalette::ColorGroup::Normal : QPalette::ColorGroup::Disabled,
+					QPalette::WindowText);
+
+				QTableWidgetItem* item1 = bpTableWidget->item(bpRow, ENABLED);
+				item1->setCheckState(Qt::Checked);
+
+				QTableWidgetItem* item2 = bpTableWidget->item(bpRow, BP_ADDRESS);
+				item2->setForeground(color);
+				item2->setText(hexValue(bp.address, 4));
+
+				QTableWidgetItem* item3 = bpTableWidget->item(bpRow, CONDITION);
+				item3->setForeground(color);
+				item3->setText(bp.condition.simplified());
+
+				QTableWidgetItem* item4 = bpTableWidget->item(bpRow, SLOT);
+				item4->setForeground(color);
+				item4->setText(QString("%1/%2")
+					.arg(bp.ps == -1 ? QChar('X') : QChar('0' + bp.ps))
+					.arg(bp.ss == -1 ? QChar('X') : QChar('0' + bp.ss)));
+
+				QTableWidgetItem* item5 = bpTableWidget->item(bpRow, SEGMENT);
+				item5->setForeground(color);
+				item5->setText(bp.segment == -1 ? "X" : QString("%1").arg(bp.segment));
+
+				items.append({BREAKPOINT, bpTableWidget->row(item5)});
+				break;
+			}
+			case Breakpoint::WATCHPOINT_MEMREAD:
+			case Breakpoint::WATCHPOINT_MEMWRITE:
+			case Breakpoint::WATCHPOINT_IOREAD:
+			case Breakpoint::WATCHPOINT_IOWRITE: {
+				wpRow = createTableRow(WATCHPOINT);
+				setBreakpointStatus(WATCHPOINT, wpRow, true);
+
+				bool currentSlot = (bp.type == Breakpoint::WATCHPOINT_MEMREAD
+				                 || bp.type == Breakpoint::WATCHPOINT_MEMWRITE)
+				                  ? breakpoints->isWatchpoint(bp.address) : true;
+				QBrush color = palette().brush(runState | currentSlot ?
+					QPalette::ColorGroup::Normal : QPalette::ColorGroup::Disabled,
+					QPalette::WindowText);
+
+				QTableWidgetItem* item1 = wpTableWidget->item(wpRow, ENABLED);
+				item1->setCheckState(Qt::Checked);
+
+				auto model = wpTableWidget->model();
+				auto combo = (QComboBox*) wpTableWidget->indexWidget(model->index(wpRow, WP_TYPE));
+				combo->setCurrentIndex(static_cast<int>(bp.type) - 1);
+				int adrLen = (bp.type == Breakpoint::WATCHPOINT_IOREAD
+				           || bp.type == Breakpoint::WATCHPOINT_IOWRITE) ? 2 : 4;
+
+				QString address = hexValue(bp.address, adrLen);
+				QString regionEnd = hexValue(bp.regionEnd, adrLen);
+				QTableWidgetItem* item2 = wpTableWidget->item(wpRow, WP_REGION);
+				item2->setForeground(color);
+				item2->setText(QString("%1%2%3").arg(address)
+						.arg(regionEnd == -1 || regionEnd == address ? "" : ":")
+						.arg(regionEnd == -1 || regionEnd == address ? "" : regionEnd));
+
+				const QString& condition = bp.condition;
+				QTableWidgetItem* item3 = wpTableWidget->item(wpRow, CONDITION);
+				item3->setForeground(color);
+				item3->setText(condition);
```

Comment:
Very much a detail.
Just use `bp.condition` here, and remove the local variable?